src/tools/sssctl/sssctl_data.c

```diff
@@ -290,3 +291,56 @@ errno_t sssctl_cache_upgrade(struct sss_cmdline *cmdline,
 
     return EOK;
 }
+
+errno_t sssctl_cache_expire(struct sss_cmdline *cmdline,
+                            struct sss_tool_ctx *tool_ctx,
+                            void *pvt)
+{
+    errno_t ret;
+    char *cmd_args = NULL;
+    const char *cachecmd = SSS_CACHE;
+    char *cmd = NULL;
+    int i;
+
+    if (cmdline->argc == 0) {
+        fprintf(stderr, _("Missing option. \n"));
+        cachecmd = "sss_cache --help";
+        sssctl_run_command(cachecmd);
+        ret = EXIT_FAILURE;
+        goto done;
+    }
+
+    cmd_args = talloc_strdup(tool_ctx, "");
+    if (cmd_args == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    for (i = 0; i < cmdline->argc; i++) {
+        cmd_args = talloc_strdup_append(cmd_args, cmdline->argv[i]);
+        if (i != cmdline->argc - 1) {
+            cmd_args = talloc_strdup_append(cmd_args, " ");
+        }
+    }
+
+    cmd = talloc_asprintf(tool_ctx, "%s %s", cachecmd, cmd_args);
+    if (cmd == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    printf(_("Executing command [%s]\n"), cmd);
```

Comment:
I would personally hide that we execute the sss_cache command and just remove the printf line. The user doesn't care whether another command is executed and the point is to over time hide and deprecate the other commands except for sssctl.